### ISSUE TITLE
parseJson(string) should fail on leftover data

### ISSUE THREAD

**Issue by nigredo-tori** [nim-lang/Nim#5134]
At the moment `parseJson` reads the first JSON node, and ignores the rest of the input. For example:
```nim
echo parseJson("123 456")  # Prints "123"
```
This behavior is counter-intuitive (at least for the string version / `parseFile`) and isn't mentioned in documentation. I'd expect parsing the whole input to be a default, with an option to drop down to streams for more control. As it stands, just to check if a string is valid JSON I have to dive into streams.

1. If this behavior is deliberate - it should be mentioned in documentation.
2. If it is not deliberate - it should be corrected (and the new behavior mentioned in documentation).

For an example of how this is implemented in other languages - `json.loads` in Python considers this an error:
```
json.decoder.JSONDecodeError: Extra data: line 1 column 5 (char 4)
```

PS: Actually, I can't even check whether a string is valid JSON using streams, since lexbase does some internal buffering.

**Comment by Araq** [comment]
It's just a bug.

**Comment by loloiccl** [comment]
This issue should be closed.
I should write not "Fix for #5134" but "Fix #5134" in commit message.
Would anyone close this issue?
